src/component.jl

```diff
@@ -302,3 +302,201 @@ function _shared_component(typedef, mod)
     end
 end
 
+
+################################################################################
+
+struct ApplyToGroup
+    e::Entity
+end
+
+Base.parent(e::Entity) = ApplyToGroup(e)
+
+struct GroupedComponent{T <: ComponentData} <: AbstractComponent{T}
+    indices::Indices
+    group::Vector{Int}
+    group_size::Vector{Int}
+    data::Vector{T}
+end
+
+GroupedComponent{T}() where {T <: ComponentData} = GroupedComponent{T}(Indices(), Int[], Int[], T[])
+
+Base.@propagate_inbounds @inline Base.getindex(c::GroupedComponent, e::Entity) = c.data[c.group[c.indices[e.id]]]
+Base.@propagate_inbounds @inline Base.getindex(c::GroupedComponent, i::Integer) = c.data[c.group[i]]
+
+function is_unique_in(value, collection)
+    count = 0
+    for element in collection
+        count += element == value
+    end
+    return count == 1
+end
+
+
+# c[entity] = value
+# set value of <only> this entity
+@inline function Base.setindex!(c::GroupedComponent{T}, v::T, e::Entity) where {T}
+    eid = e.id
+    @inbounds if in(e, c)
+        g = c.group[c.indices[eid]]
+        if c.group_size[g] == 1 #is_unique_in(c.group[c.indices[eid]], c.group)
+            # the entity already has its own group - adjust value
+            c.data[g] = v
+        else
+            # the entity is part of a group - create a new one
+            c.group_size[g] -= 1
+            push!(c.data, v)
+            push!(c.group_size, 1)
+            c.group[c.indices[eid]] = length(c.data)
+        end
+    else
+        # the entity is not in the component - add it
+        push!(c.indices, eid)
+        push!(c.group, length(c.group)+1)
+        push!(c.group_size, 1)
+        push!(c.data, v)
+    end
+    return v
+end
+
+# c[entity] = parent
+# set the value of this entity to that of parent
+@inline function Base.setindex!(c::GroupedComponent, p::Entity, e::Entity)
+    @boundscheck if !in(p, c)
+        throw(BoundsError(c, p))
+    end
+    pg = c.group[c.indices[p.id]]
+    if in(e, c)
+        eg = c.group[c.indices[e.id]]
+        if c.group_size[eg] == 1 #is_unique_in(c.group[c.indices[eid]], c.group)
+            # if this entity is the only one holding onto a value, remove that 
+            # value and cleanup group indices
+            deleteat!(c.data, idx)
+            deleteat!(c.group_size, idx)
+            for i in eachindex(c.group)
+                c.group[i] = c.group[i] - (c.group[i] > eg)
```

Comment:
cool, I did not know this - Bool turns into - 0 or 1 :D

Reply:
Yea, you can skip branching with that sometimes